agent_config.go

```diff
@@ -100,6 +103,11 @@ type AgentConfig struct {
 	// keepAlive behavior should be modified with KeepaliveInterval and ConnectionTimeout
 	taskLoopInterval time.Duration
 
+	// taskLoopInterval controls how often our internal task loop runs when in the connecting phase,
+	// this would normally be more frequent thank the normal "taskLoopInterval", to allow
+	// quick state changes and connection establishment
+	TaskLoopConnectingInterval time.Duration
```

Comment:
IMO, `TaskLoop` would be better not to be appeared in public API as it is from internal implementation. (It can also be implemented by `time.Timer` without having a loop.)
It may be something like `ConnectingPingInterval`?